api/src/main/java/org/openmrs/api/handler/PatientSaveHandler.java

```diff
@@ -45,5 +46,10 @@ public void handle(Patient patient, User creator, Date dateCreated, String other
 				}
 			}
 		}
+
+		if(patient.getPatientId() == null){
+			patient.setCreator(Context.getAuthenticatedUser());
```

Comment:
Is this a bug fix? Or is it a change caused by switching to annotations?

Reply:
Not really but sort of, more like bad design, this is needed because creator and dateCreated fields exist both in  BaseOpenmrsData and Patient, this is actually bad practice because it defeats the purpose of inheritance, and this where xml and annotations differ, because annotations are pure java whereas xml isn't, with xml you get away with bad things just like we were doing before which is why I say sort of a bug.

Anyways, long story short, hibernate is unable to set these fields via the AuditableInterceptor mechanism because based on the ClassMedata it has built for Patient class, it sees the same fields on the superclass and the subclass and treats them as different because they are declared by different classes anyways, it only sets the values of the fields in the superclass and not the subclass so we have to manually set them from here as we used to do before to get around it.

Reply:
What do you think of adding this comment to the code?